src/utils/formatSearchResponse.ts

```diff
@@ -54,15 +54,86 @@ const formatSearchResponse = async (
+  }
+  if (searchItem?._source?.MD_LegalConstraintsOtherConstraintsObject?.[2]?.text) {
+    licenseConstraints =
+      licenseConstraints + `${searchItem?._source?.MD_LegalConstraintsOtherConstraintsObject?.[2]?.text}`;
+  }
+
+  return licenseConstraints;
+};
+
+const getLimitationPublicAccess = (searchItem: Record<string, any>): string => {
+  let limitationPublicAccess = '';
+
+  if (searchItem?._source?.cl_accessConstraints?.[0]?.default) {
+    limitationPublicAccess = `${searchItem?._source?.cl_accessConstraints?.[0]?.default} <br>`;
+  }
+  if (searchItem?._source?.cl_accessConstraints?.[0]?.text) {
+    limitationPublicAccess = limitationPublicAccess + `${searchItem?._source?.cl_accessConstraints?.[0]?.text} <br>`;
+  }
+  if (searchItem?._source?.licenseObject?.[0]?.defasult) {
+    limitationPublicAccess = limitationPublicAccess + `${searchItem?._source?.licenseObject?.[0]?.default}`;
+  }
+
+  return limitationPublicAccess;
+};
+
+const getPublishedBy = (publishedBy: Record<string, any>): string => {
+  let dataOwner = '';
+
+  if (publishedBy.role) {
+    dataOwner = `${publishedBy.role}, `;
+  }
+  if (publishedBy.organisationValue) {
+    dataOwner += `${publishedBy.organisationValue} <br>`;
+  }
+  if (publishedBy.email) {
+    dataOwner += `${publishedBy.email}`;
+  }
+
+  return dataOwner;
+};
+
+const getCatalogueDate = (searchItem: Record<string, any>): string => {
   const catalogueDate = searchItem?._source?.dateStamp ? new Date(searchItem?._source?.dateStamp) : '';
+  return catalogueDate
+    ? catalogueDate.toLocaleDateString('en-US', { day: 'numeric' }) +
+        '-' +
+        catalogueDate.toLocaleDateString('en-US', { month: 'short' }) +
+        '-' +
+        catalogueDate.toLocaleDateString('en-US', { year: 'numeric' })
+    : '';
+};
+
+const getKeywords = (searchItem: Record<string, any>): string => {
+  return searchItem?._source?.tag?.map((item) => item.default).join(', ') ?? '';
+};
+
+const getTopicCategories = (searchItem: Record<string, any>): string => {
+  return searchItem?._source?.cl_topic?.map((item) => item.default).join(', ') ?? '';
+};
+
+const getHostCatalogueNumber = (searchItem: Record<string, any>): string => {
+  return `${searchItem?._source?.resourceIdentifier?.[0]?.codeSpace ?? ''} ${searchItem?._source?.resourceIdentifier?.[0]?.code ?? ''}`;
+};
+
+const getOtherDetails = async (
+  searchItem: Record<string, any>,
+  publishedBy: Record<string, any>,
+): Promise<Record<string, string>> => {
   return {
     alternateTitle: searchItem?._source?.resourceAltTitleObject?.[0]?.default ?? '',
     language: searchItem?._source?.mainLanguage?.toUpperCase() ?? '',
-    keywords: searchItem?._source?.tag?.map((item) => item.default).join(', ') ?? '',
-    topic_categories: searchItem?._source?.cl_topic?.map((item) => item.default).join(', ') ?? '',
+    keywords: getKeywords(searchItem),
+    topic_categories: getTopicCategories(searchItem),
```

Comment:
I have already made changes to this in my open [PR](https://github.com/DEFRA/ncea-frontend/pull/34).

Reply:
I had to do this because of the sonar error.